WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift

```diff
@@ -256,11 +262,23 @@ private extension DashboardViewController {
         }
     }
 
-    func reloadDashboardUIStatsVersion(forced: Bool) {
-        dashboardUIFactory.reloadDashboardUI(onUIUpdate: { [weak self] dashboardUI in
+    func reloadDashboardUIStatsVersion(forced: Bool) async {
```

Comment:
I was having crashes when loading the dashboard due to this method being called in the background. I applied @toupper's suggestion above but it didn't work, so I think we should instead probably mark this method with `@MainActor` instead to enforce that this method is called on the main thread:

```suggestion
    @MainActor func reloadDashboardUIStatsVersion(forced: Bool) async {
```

Reply:
Very interesting that you got crashes, I haven't seen any crashes on my end with Xcode 13.3 🤔 do you have the steps to repro, and what's the Xcode version that you use?

Reply:
I just tested the branch with Xcode 13.2.1, and it's now crashing consistently! it's good to be able to reproduce, maybe something changed in Xcode 13.3 that always calls functions in  `Task` inside a `@MainActor` class like a `UIViewController` subclass on the main thread. I'll look into this more!

Reply:
Yes, I'm using Xcode 13.2.1 too.

Reply:
Also no crashes on my side with XCode 13.3 😅 

> I was having crashes when loading the dashboard due to this method being called in the background. I applied @toupper's suggestion above but it didn't work, so I think we should instead probably mark this method with @MainActor instead to enforce that this method is called on the main thread:

Oh I am wondering why `Task { @MainActor in ...} `didn't work, that is supposed to hop to the main thread to run the code in the closure. When marking functions and classes with @MainActor we have to be super careful, because if the caller is  non-concurrency-aware code, it will do nothing 😱.
There is a very good [post](https://mobile.blog/2022/03/28/swift-why-is-my-mainactor-code-running-in-the-background/#:~:text=By%20marking%20a%20function%20with,it%20on%20the%20main%20thread.) from bjhomer that explains this one, also [here](https://www.swiftbysundell.com/articles/the-main-actor-attribute/). I checked the code in this PR and all seem to be called from concurrency-aware code, but It would be worth it to double-check.
If still the risks are too big we could use `DispatchQueue.main.async()` for now, merge the PR, and revisit it in another issue 🚀

Reply:
@toupper I believe that this was crashing and `Task { @MainActor in ...}` was not working only because it was in Xcode 13.2, confirmed in [BJ's comment](pbMoDN-3mQ-p2#comment-6748) - there is [a bug in the Swift optimizer](https://github.com/apple/swift/issues/58066) that causes some `@MainActor` things to run in the background. He suggested using Xcode 13.3+ for Swift concurrency usage, and the [CI is on Xcode 13.3.1](https://github.com/woocommerce/woocommerce-ios/pull/6884).

I also asked in the p2 discussion whether we should still add `@MainActor` to functions/properties in a `@MainActor` class since they should already be run on the main thread after the bug fix in Xcode 13.3. I still updated the PR to add `@MainActor` to the relevant functions and `Task` just to be extra safe, and for team members who are still using Xcode 13.2.

Reply:
> @toupper I believe that this was crashing and Task { @MainActor in ...} was not working only because it was in Xcode 13.2, confirmed in [BJ's comment](https://github.com/woocommerce/woocommerce-ios/pull/6890/pbMoDN-3mQ-p2#comment-6748) - there is https://github.com/apple/swift/issues/58066 that causes some @MainActor things to run in the background. He suggested using Xcode 13.3+ for Swift concurrency usage, and the https://github.com/woocommerce/woocommerce-ios/pull/6884.

@jaclync Thanks for the info! That makes total sense, on Xcode 13.3 it worked fine for me. I am going to start an async discussion on whether we should update our Xcode locally so we can align to the Ci and use the new Swift Concurrency with its full power 😃

Reply:
Cool thanks for the announcement in the async channel!